utils/das_data_sorter.py

```diff
@@ -128,43 +129,97 @@ def gps_data(self, data:pd.Series) -> pd.Series:
+                self.data[variable] = self.__median_smooth(self.data[variable], n)
+
+    def __mean_smooth(self, data:pd.Series, n:int) -> pd.Series:
+        '''Returns an array of smoothed data based on number of data points taken to smooth. 
+        
+        For an odd number N, the data points are simply averaged. 
+        Whereas for an even number N, the data points are averaged, then centered. This is because the data point will misalign with integer
+        numbers of time if not done.'''
+        smooth_data_array = [] 
+
+        if n % 2 != 0: # Mean Smoothing for odd number N
+            for i in range(len(data) - n + 1):
+                data_points = data[i:i+n]
+                new_data_point = self.mean(data_points)
+                new_data_point = round(new_data_point, ndigits=2)
+                smooth_data_array.append(new_data_point)
+        else: # Mean Smoothing for even number N
+            temp_array = []
+            for i in range(len(data) - n + 1):
+                data_points = data[i:i+N]
+                new_data_point = self.mean(data_points)
+                temp_array.append(new_data_point)
+            
+            # an extra step to centre the data by averaging adjacent data points
+            for j in range(len(temp_array) - 1): 
+                first = temp_array[j]
+                second = temp_array[j+1]
+                new_data_point = self.mean([first,second])
+                new_data_point = round(new_data_point, ndigits=2)
+                smooth_data_array.append(new_data_point)
+        
+        return smooth_data_array 
+    
+    def __median_smooth(self, data:pd.Series, n:int) -> pd.Series:
+        '''Returns an array of smoothed data based on number of data points taken to smooth. 
+        
+        For an odd number N, the data points are simply averaged. 
+        Whereas for an even number N, the data points are averaged, then centered. This is because the data point will misalign with integer
+        numbers of time if not done.'''
+        smooth_data_array = [] 
+
+        if n % 2 != 0: # Median Smoothing for odd number N
+            for i in range(len(data) - n + 1):
+                data_points = data[i:i+n]
+                new_data_point = median(data_points) # round not required, since it will always be the middle number
+                smooth_data_array.append(new_data_point)
+        else: # Median Smoothing for even number N
+            temp_array = []
+            for i in range(len(data) - n + 1):
+                data_points = data[i:i+n]
+                new_data_point = median(data_points)
+                temp_array.append(new_data_point)
+            
+            # an extra step to centre the data by averaging adjacent data points
+            for j in range(len(temp_array) - 1):
+                first = temp_array[j]
+                second = temp_array[j+1]
+                new_data_point = median([first,second])
```

Comment:
I think finding the mean here would be more clear than finding the median, even though it gives the same result for a two element array. Just cos when I think of what operation is used to find the value exactly half way between two others, my mind always goes to mean rather than median.

Reply:
I don't mind doing this, I was just thinking whether changing this to `self.mean` for a Median technique would be confusing for future Electrical team who are learning about 'Moving Average' techniques for the first time. Would it be confusing or am I overthinking this?

@khanguslee @minh-tn would like your opinions about this as well

Reply:
I think for a person learning about how the maths works, the most confusing point would be why we even do this second round of averaging to center the data. If they've figured that out, I think using mean wouldn't be any more confusing than median. Besides, I reckon using mean better describes what we're trying to do, so it could be less confusing!